processor/src/chiplets/bus/mod.rs

```diff
@@ -39,32 +41,35 @@ impl ChipletsBus {
     /// Requests lookups for a single operation at the specified cycle. A Hasher operation request
     /// can contain one or more lookups, while Bitwise and Memory requests will only contain a
     /// single lookup.
-    fn request_lookup(&mut self, request_cycle: u32) {
-        // All requests are sent from the stack. Requests are guaranteed not to share cycles with
-        // other requests, but they might share a cycle with a response which has already been
-        // sent.
+    fn request_lookup(&mut self, request_cycle: Cycle) {
         let request_idx = self.request_rows.len();
         self.lookup_hints
             .entry(request_cycle)
-            .and_modify(|lookup| {
-                if let ChipletsLookup::Response(response_idx) = *lookup {
-                    *lookup = ChipletsLookup::RequestAndResponse((request_idx, response_idx));
+            .and_modify(|lookup| match lookup {
+                // Requests might share a cycle with a response which has already been sent.
+                ChipletsLookup::Response(response_idx) => {
+                    *lookup = ChipletsLookup::RequestAndResponse((request_idx, *response_idx));
                 }
+                // Requests are guaranteed not to share cycles with other requests.
+                _ => debug_assert!(false, "bus already contains a Request"),
             })
             .or_insert_with(|| ChipletsLookup::Request(request_idx));
     }
 
     /// Provides lookup data at the specified cycle, which is the row of the Chiplets execution
     /// trace that contains this lookup row.
-    fn provide_lookup(&mut self, response_cycle: u32) {
-        // results are guaranteed not to share cycles with other results, but they might share
-        // a cycle with a request which has already been sent.
+    fn provide_lookup(&mut self, response_cycle: Cycle) {
         let response_idx = self.response_rows.len();
         self.lookup_hints
             .entry(response_cycle)
             .and_modify(|lookup| {
-                if let ChipletsLookup::Request(request_idx) = *lookup {
-                    *lookup = ChipletsLookup::RequestAndResponse((request_idx, response_idx));
+                match lookup {
+                    // Results might share a cycle with a request which has already been sent.
+                    ChipletsLookup::Request(request_idx) => {
+                        *lookup = ChipletsLookup::RequestAndResponse((*request_idx, response_idx));
+                    }
+                    // Results are guaranteed not to share cycles with other results.
+                    _ => debug_assert!(false, "bus already contains a Request"),
```

Comment:
nit: As long as we're improving these comments, we should probably also switch to referring to this as a "response" rather than a "result"